### PR TITLE
Fix "exports is not defined" error

### PR DESCRIPTION
**Pull request by PVince81**
Adjust webpack config to accomodate for the updates of @babel/preset-env from https://github.com/nextcloud/notifications/pull/899

Fixes https://github.com/nextcloud/server/issues/26314

Inspired by a similar issue we had on the talk repo: https://github.com/nextcloud/spreed/pull/4829/commits/5673db93aee8b85a0344dff33454e3d7898b6f50#diff-3e31f6f375484b2dfd4629faed9e5183ac6568ebdbc67ec752e10c41a35250c6R72

### CODE REVIEW

**Review by MorrisJobke — APPROVED** [review]
🐘

**Review by nickvergessen — APPROVED** [review]

**Comment by nickvergessen** [comment]
Seems to work and we do the same in talk, but I'd still like confirmation by another JS expert

File: webpack.js [hunk 1]
```diff
@@ -8,6 +8,17 @@ webpackConfig.module.rules[jsRule].exclude = BabelLoaderExcludeNodeModulesExcept
...
 ])
+webpackConfig.module.rules[jsRule].options = {
+	plugins: ['add-module-exports'],
+	presets: [
+		/**
+		 * From "add-module-exports" documentation:
+		 * "webpack doesn't perform commonjs transformation for
+		 * codesplitting. Need to set commonjs conversion."
+		 */
+		['@babel/env', { modules: 'commonjs' }],
```
Comment (skjnldsv) [inline webpack.js:19]: I would prefer that in the babelrc, no?

Comment (PVince81) [reply webpack.js:19]: @skjnldsv any quick hints how to do that ? I'm doing this just as a quick fix to get server master CI unblocked

Comment (PVince81) [reply webpack.js:19]: okay, I found something https://babeljs.io/docs/en/configuration#babelrcjson

but I'm unsure why these options are attached to a "jsRule" in webpack and whether they'll work the same if I specify the plugin + presets directly in the RC file

Comment (PVince81) [reply webpack.js]: ok, got it working. 

@skjnldsv please review last commit

**Comment by PVince81** [comment]
CI is green. I've also retested this PR's state after checking it out again and it works.

@skjnldsv are you ok with the babelrc changes ?

**Comment by nickvergessen** [comment]
Merge with `babel.config.js` please

File: .babelrc [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (nickvergessen) [inline .babelrc:1]: Merge with `babel.config.js` please

**Review by nickvergessen — CHANGES_REQUESTED** [review]
see above

**Comment by PVince81** [comment]
done. merged the configs, tested then squashed

**Review by nickvergessen — APPROVED** [review]

**Comment by PVince81** [comment]
I did run `make dev-setup && make build-js-production` every time, and yet it doesn't accept my compiled assets.

**Comment by skjnldsv** [comment]
I wonder if we could have a global babel config too.
Would be nice

**Comment by PVince81** [comment]
> I wonder if we could have a global babel config too.
> Would be nice

I wonder too, but won't be delivered today.
There's a master branch that is waiting to be unblocked: https://github.com/nextcloud/server/issues/26314

**Comment by skjnldsv** [comment]
> I wonder too, but won't be delivered today.

Let me test

**Pull request merged**